src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -45,464 +46,505 @@ public static IEnumerable<object[]> Match_Basic_TestData_LazyLoops()
+                yield return new object[] { engine, @"\w{0,2}c@\w+.com", "abc@def.com", RegexOptions.None, 0, 11, true, "abc@def.com" };
+                yield return new object[] { engine, @"\w*@\w+.com", "abc@def.com", RegexOptions.None, 0, 11, true, "abc@def.com" };
+                yield return new object[] { engine, @"(\w+)@\w+.com", "abc@def.com", RegexOptions.None, 0, 11, true, "abc@def.com" };
+                yield return new object[] { engine, @"((\w+))@\w+.com", "abc@def.com", RegexOptions.None, 0, 11, true, "abc@def.com" };
+                yield return new object[] { engine, @"(\w+)c@\w+.com", "abc@def.comabcdef", RegexOptions.None, 0, 17, true, "abc@def.com" };
+                if (!RegexHelpers.IsNonBacktracking(engine))
+                {
+                    yield return new object[] { engine, @"(\w+)c@\w+.com\1", "abc@def.comabcdef", RegexOptions.None, 0, 17, true, "abc@def.comab" };
+                    yield return new object[] { engine, @"(\w+)@def.com\1", "abc@def.comab", RegexOptions.None, 0, 13, false, string.Empty };
+                    yield return new object[] { engine, @"(\w+)@def.com\1", "abc@def.combc", RegexOptions.None, 0, 13, true, "bc@def.combc" };
+                    yield return new object[] { engine, @"(\w*)@def.com\1", "abc@def.com", RegexOptions.None, 0, 11, true, "@def.com" };
+                    yield return new object[] { engine, @"\w+(?<!a)", "a", RegexOptions.None, 0, 1, false, string.Empty };
+                    yield return new object[] { engine, @"\w+(?<!a)", "aa", RegexOptions.None, 0, 2, false, string.Empty };
+                    yield return new object[] { engine, @"(?>\w+)(?<!a)", "a", RegexOptions.None, 0, 1, false, string.Empty };
+                    yield return new object[] { engine, @"(?>\w+)(?<!a)", "aa", RegexOptions.None, 0, 2, false, string.Empty };
+                }
+                yield return new object[] { engine, @".+a", "baa", RegexOptions.None, 0, 3, true, "baa" };
+                yield return new object[] { engine, @"[ab]+a", "cacbaac", RegexOptions.None, 0, 7, true, "baa" };
+                yield return new object[] { engine, @"^(\d{2,3}){2}$", "1234", RegexOptions.None, 0, 4, true, "1234" };
+                yield return new object[] { engine, @"(\d{2,3}){2}", "1234", RegexOptions.None, 0, 4, true, "1234" };
+                yield return new object[] { engine, @"((\d{2,3})){2}", "1234", RegexOptions.None, 0, 4, true, "1234" };
+                if (!RegexHelpers.IsNonBacktracking(engine))
+                {
+                    // TODO-NONBACKTRACKING: Produce "1234" rather than "123", which could be considered an alternative valid result given greedy semantics
+                    yield return new object[] { engine, @"(\d{2,3})+", "1234", RegexOptions.None, 0, 4, true, "123" };
+                    yield return new object[] { engine, @"(\d{2,3})*", "123456", RegexOptions.None, 0, 4, true, "123" };
```

Comment:
@veanes, @olsaarik, the NonBacktracking engine is producing a different answer here than the other engines.  I assume we'd consider its answer to be valid, but wanted to double-check it's intentional.

Reply:
fixed in #1626